lib/src/layer.dart

```diff
@@ -67,12 +70,18 @@ class Layer {
         tileIndex += 4;
 
         // Read out the flags
-        var flipped_horizontally = (globalTileId & FLIPPED_HORIZONTALLY_FLAG);
-        var flipped_vertically = (globalTileId & FLIPPED_VERTICALLY_FLAG);
-        var flipped_diagonally = (globalTileId & FLIPPED_DIAGONALLY_FLAG);
+        bool flippedHorizontally = (globalTileId & FLIPPED_HORIZONTALLY_FLAG) == FLIPPED_HORIZONTALLY_FLAG;
+        bool flippedVertically = (globalTileId & FLIPPED_VERTICALLY_FLAG) == FLIPPED_VERTICALLY_FLAG;
+        bool flippedDiagonally = (globalTileId & FLIPPED_DIAGONALLY_FLAG) == FLIPPED_DIAGONALLY_FLAG;
 
-        // Clear the flags
+        // Save rotation flags
+        tileRotation[y][x] = [
+          flippedHorizontally,
+          flippedVertically,
+          flippedDiagonally
+        ];
```

Comment:
Let's create a class `Flips` that contains three distinct boolean fields? `horizontal`, `vertical` and `diagonal`? It's hard to know what the elements of the array mean when looking at that on other parts of the code.
Then we can have a List<List<Flips>> on the Layer